src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -401,11 +408,13 @@ public function setSaveHandler($saveHandler = null)
         if (!$saveHandler instanceof AbstractProxy && $saveHandler instanceof \SessionHandlerInterface) {
             $saveHandler = new SessionHandlerProxy($saveHandler);
         } elseif (!$saveHandler instanceof AbstractProxy) {
-            $saveHandler = new SessionHandlerProxy(new \SessionHandler());
+            $saveHandler = new SessionHandlerProxy(new StrictSessionHandler(new \SessionHandler()));
```

Comment:
It feels strange that we need to make the native session andlers strict. The `files` session handler is already strict I assume. So this is only needed because if you use `\SessionHandler`, you lose that strictness? \SessionHandler does not implement `\SessionUpdateTimestampHandlerInterface`! So if we not not set a save handler at all it would even be better as it would then use the validate id implementation of the native session handler? But currently that is not possible do.

Reply:
Correct, I don't think we can do better here, id validation works with the wrapper so not a big issue. There is only updateTimestamp which ships no optimization but only writes. Not sure we can do better.

Reply:
If php doesn't change this behavior, we can think about deprecating NativeFilesSessionHandler in SF 4.1. So by default it just sets the session handler without actually overwriting \SessionHandler.